### PR TITLE
MGDOBR-794: Add azure event hubs action

### PR DESCRIPTION
**Pull request by r00ta**
[MGDOBR-794](https://issues.redhat.com/browse/MGDOBR-794)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

**Review by danielezonca — APPROVED** [review]

**Comment by manstis** [comment]
Is the title of this PR wrong?

It seems to duplicate (in part?) https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/986

**Review by kelvah — APPROVED** [review]

**Comment by Aakanksha0311** [comment]
The PR looks good to me. Checked now.

**Review by Aakanksha0311 — APPROVED** [review]

**Comment by r00ta** [comment]
In theory MC fixed the connector, shall we give it another try @sutaakar ? Just ping me when you have time to test this so that I can allocate the resources if you need

File: processor/configuration/src/main/resources/schemas/actions/azure_eventhubs_sink_0.1.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        "description" : "The key for EventHubs SAS key name",
+        "type" : "string",
+        "format" : "password"
+      }, {
+        "description" : "An opaque reference to the azure_shared_access_key",
+        "type" : "object",
+        "properties" : { }
+      } ]
+  }
+}
```
Comment (kelvah) [inline processor/configuration/src/main/resources/schemas/actions/azure_eventhubs_sink_0.1.json:41]: It looks like there's a missing closing bracket here?

**Comment by sutaakar** [comment]
@Aakanksha0311 will check that again, last month it was working properly.

File: processor/configuration/src/main/resources/schemas/actions/azure_eventhubs_sink_0.1.json (hunk 1 above)
Comment (r00ta) [reply processor/configuration/src/main/resources/schemas/actions/azure_eventhubs_sink_0.1.json:45]: Thanks good spot :+1: The pipeline was failing indeed :sweat_smile:

**Comment by Aakanksha0311** [comment]
> @Aakanksha0311 will check that again, last month it was working properly.

Yes, checked and LGTM.

**Review by Aakanksha0311 — APPROVED** [review]

**Review by Aakanksha0311 — APPROVED** [review]

File: processor/configuration/src/test/java/com/redhat/service/smartevents/processor/validators/DefaultGatewayValidatorTest.java [hunk 2]
```diff
@@ -183,6 +184,24 @@ void testKafkaTopicAction() {
         assertValidationIsValid(validKafkaTopicAction);
     }
 
+    @Test
+    void testAzureEventHubAction() {
+        Map<String, String> invalidParams = new HashMap<>();
+        invalidParams.put(AzureEventHubsAction.AZURE_SHARED_ACCESS_KEY_PARAM, "example");
```
Comment (jomarko) [inline processor/configuration/src/test/java/com/redhat/service/smartevents/processor/validators/DefaultGatewayValidatorTest.java:190]:
```suggestion
        Map<String, String> invalidParams = java.util.Collections.singletonMap(AzureEventHubsAction.AZURE_SHARED_ACCESS_KEY_PARAM, "example");
```

File: processor/configuration/src/main/resources/schemas/actions/azure_eventhubs_sink_0.1.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+          "type": "string",
+          "format": "password"
+        },
+        {
+          "description": "An opaque reference to the azure_shared_access_key",
+          "type": "object",
+          "properties": {}
+        }
+      ]
+    }
```
Comment (jomarko) [inline processor/configuration/src/main/resources/schemas/actions/azure_eventhubs_sink_0.1.json:43]: we use two versions EventHubs and eventhubs, probably we should unify the name we use. Or do we refer by both to a different thing?

**Review by jomarko — COMMENTED** [review]
Thank you for a PR. some minor comments. one of them is optional code shortening.

File: processor/configuration/src/main/resources/schemas/actions/azure_eventhubs_sink_0.1.json (hunk 3 above)
Comment (r00ta) [reply processor/configuration/src/main/resources/schemas/actions/azure_eventhubs_sink_0.1.json:43]: wdym?

File: processor/configuration/src/test/java/com/redhat/service/smartevents/processor/validators/DefaultGatewayValidatorTest.java (hunk 2 above)
Comment (r00ta) [reply processor/configuration/src/test/java/com/redhat/service/smartevents/processor/validators/DefaultGatewayValidatorTest.java:190]: I would refactor this in another PR if it's ok by you

File: processor/configuration/src/main/resources/schemas/actions/azure_eventhubs_sink_0.1.json (hunk 3 above)
Comment (jomarko) [reply processor/configuration/src/main/resources/schemas/actions/azure_eventhubs_sink_0.1.json:43]: I mean this small difference in naming
- "The eventhubs namespace",
- "EventHubs SAS key name"

What is your opinion of using 'eventhubs' vs 'EventHubs'

**Pull request merged**